pep-0696.rst

```diff
@@ -265,9 +276,11 @@ The constraints of ``T2`` must be a superset of the constraints of ``T1``.
    class Bar(Generic[T, ListDefaultT]):
        def __init__(self, x: T, y: ListDefaultT): ...
 
-   reveal_type(Bar[int])             # type is Bar[int, list[int]]
-   reveal_type(Bar[int, list[str]])  # type is Bar[int, list[str]]
-   reveal_type(Bar[int, str])        # type is Bar[int, str]
+   reveal_type(Bar)                    # type is type[Bar[T, list[ListDefaultT = T]]]
+   reveal_type(Bar[int])               # type is type[Bar[int, ListDefaultT = list[int]]]
```

Comment:
I think you're saying that partial specialization is applied in this case? This is not what I'd expect. I think that when the use indicates an intent to specialize a generic type by using an index expression form, it should be fully specialized, with defaults applied as necessary. In other words, I think the revealed type here should be `type[Bar[int, list[int]]`.

Allowing partial specialization here would be inconsistent with existing rules for specialization where all TypeVar's have implied default types of "Any". Consider, for example, `reveal_type(dict[str])`. This should reveal `dict[str, Any]`, not `dict[str, _VT = Any]`.

Reply:
Yeah that makes sense, I've fixed the examples to remove examples of this